src/main/java/com/yahoo/validatar/execution/pig/Sty.java

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright 2014-2015 Yahoo! Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.yahoo.validatar.execution.pig;
+
+import com.yahoo.validatar.common.Helpable;
+import com.yahoo.validatar.common.Query;
+import com.yahoo.validatar.common.Result;
+import com.yahoo.validatar.common.TypeSystem;
+import com.yahoo.validatar.common.TypedObject;
+import com.yahoo.validatar.execution.Engine;
+import joptsimple.OptionParser;
+import joptsimple.OptionSet;
+import org.apache.log4j.Logger;
+import org.apache.pig.PigServer;
+import org.apache.pig.backend.executionengine.ExecException;
+import org.apache.pig.data.DataType;
+import org.apache.pig.data.Tuple;
+import org.apache.pig.impl.logicalLayer.schema.Schema;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.sql.Timestamp;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+import static java.util.Collections.singletonList;
+
+public class Sty implements Engine {
+    public static final String PIG_EXEC_TYPE = "pig-exec-type";
+    public static final String PIG_OUTPUT_ALIAS = "pig-output-alias";
+    public static final String PIG_SETTING = "pig-setting";
+
+    protected final Logger log = Logger.getLogger(getClass());
+
+    /** Engine name. */
+    public static final String ENGINE_NAME = "pig";
+
+    public static final String DEFAULT_EXEC_TYPE = "mr";
+    public static final String DEFAULT_OUTPUT_ALIAS = "validatar_results";
+    public static final String SETTING_DELIMITER = "=";
+
+    public static final String METADATA_EXEC_TYPE_KEY = "exec-type";
+    public static final String METADATA_ALIAS_KEY = "output-alias";
+
+    private String defaultExecType;
+    private String defaultOutputAlias;
+    private Properties properties;
+
+    private class FieldDetail {
+        public String alias;
+        public byte type;
+
+        public FieldDetail(String alias, byte type) {
+            this.alias = alias;
+            this.type = type;
+        }
+    }
+
+    private OptionParser parser = new OptionParser() {
+        {
+            acceptsAll(singletonList(PIG_EXEC_TYPE), "The exec-type for Pig to use.  This is the " +
+                                                       "-x argument used when running Pig. Ex: local, mr, tez etc. ")
```

Comment:
Cool, so we could execute pig script Validatar tests against sample data locally without any hadoop cluster!